pkg/generic/thrift/json_go116plus_amd64.go

```diff
@@ -106,16 +107,17 @@ func (m *WriteJSON) writeFields(ctx context.Context, out thrift.TProtocol, msgTy
 		}
 		switch msgType {
 		case Void:
-			if err := out.WriteStructBegin(field.Name()); err != nil {
-				return err
-			}
-			if err := out.WriteFieldStop(); err != nil {
-				return err
-			}
-			if err := out.WriteStructEnd(); err != nil {
+			if err := writeFieldForVoid(field.Name(), out); err != nil {
 				return err
 			}
 		case String:
+			// if the field type is void, write void instead of calling dynamicgo with string
```

Comment:
Move this logic before 'witch case'? I think it's universal for any type.

Reply:
@jayantxie Fixed. Please take a look again.